src/tests/cmocka/test_negcache.c

```diff
@@ -921,6 +924,255 @@ static void test_sss_ncache_reset_prepopulate(void **state)
     assert_int_equal(ret, EEXIST);
 }
 
+/* The main purpose of test_sss_ncache_short_name_in_domain is to test that
+ * short names in the filter_users or filter_groups options in a [domain/...]
+ * section are properly added to the related sub-domains as well (if there are
+ * any) and not added to domains from other [domain/...] sections. For
+ * completeness entries with fully-qualified names of the parent and the
+ * sub-domain and the generic UPN are added as well.
+ *
+ * The result should of course be independent of the present domains. To
+ * verify this the domains are added one after the other and the negative
+ * cache is repopulated each time.
+ *
+ * With the given domains, users and group we have to following expectations:
+ *  - the short name entry will be added to the domain and all sub-domains
+ *  - the short name expanded to the corresponding fully-qualified name of the
+ *    given domain is added to the domain and all sub-domains
```

Comment:
Again, I'm not sure if I read this correctly but what I see in the code is that "fqname" is built for every domain specifically (i.e. it's not "name expanded to ... given domain is added to ... all sub-domains"

Reply:
Also those two items doesn't mention "short name as upn".
IIUC expanded short name is added as upn to domain and all sub-domains,

Reply:
You are correct with 'What I see in the code' and my intention was to express the same in the comment. I rephrased the comment and combined the first two items into one since they meant the same.

Reply:
Thank you.